pages/profile.php

```diff
@@ -396,16 +396,15 @@
                                     </div>
 
                                     <div class="form-group">
-                                        <label class="col-sm-10 control-label"><?php echo $lang->get('timezone_selection'); ?></label>
+                                        <label class="col-sm-10 control-label"><?php echo $lang->get('timezone_selection');?></label>
                                         <div class="col-sm-10">
                                             <select class="form-control" id="profile-user-timezone">
                                                 <?php
                                                 foreach ($zones as $key => $zone) {
                                                     echo '
                                                 <option value="' . $key . '"',
-                                                    $session->has('user-timezone') && $session->get('user-timezone') && null !== $session->get('user-timezone') && $session->get('user-timezone') === $key ?
-                                                    ' selected' :
-                                                    (isset($SETTINGS['timezone']) === true && $SETTINGS['timezone'] === $key ? ' selected' : ''),
+                                                    $session->has('user-timezone') && $session->get('user-timezone') === $key ? ' selected' :
+                                                    (($session->has('user-timezone') && $session->get('user-timezone') === 'not_defined') && isset($SETTINGS['timezone']) && $SETTINGS['timezone'] === $key ? ' selected' : ''),
```

Comment:
It should work but I find that this ternary condition is complicated to read due to its length.

Reply:
Can't this be replaced by

```
$userTimezone = $session->get('user-timezone');
$selected = ($userTimezone === $key) ? ' selected' : '';

if ($userTimezone === 'not_defined' && isset($SETTINGS['timezone']) && $SETTINGS['timezone'] === $key) {
    $selected = ' selected';
}

return $selected;

```

Reply:
I agree with you, is complicated to read. I would have liked to write the code like below, but I chose to stick with the standard adopted earlier.

```php
<?php foreach ($zones as $key => $zone): ?>
    <option value="<?php echo $key; ?>"<?php 
    if ($session->has('user-timezone'))
        if($session->get('user-timezone') === $key)
            echo ' selected';
        elseif ($session->get('user-timezone') === 'not_defined')
            if (isset($SETTINGS['timezone']) && $SETTINGS['timezone'] === $key)
                echo ' selected';
    ?>><?php echo $zone; ?></option>
<?php endforeach; ?>
```

Should I commit the rewrite?

Reply:
Yes seems nice.

Can you please also sign your commit (see https://docs.github.com/fr/authentication/managing-commit-signature-verification/about-commit-signature-verification)